WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -1911,6 +1963,130 @@ static GSourceFuncs _handlerIntervention =
             return _page;
         }
 #endif
+
+
+        void collectMetricsOnLoadStart()
+        {
+
+            if (_didSendLaunchMetrics)
+               return;
+
+            auto getProcessLaunchStateString = [&]() -> std::string
+            {
+                switch(_webProcessState)
+                {
+                    case WebProcessCold: return "Cold";
+                    case WebProcessHot:  return "Hot";
+                }
+                return "Unknown";
+            };
+
+            auto addSystemInfo = [&](JsonObject &metrics)
+            {
+                struct sysinfo info;
+                if (sysinfo(&info) != 0)
+                {
+                    SYSLOG(Trace::Error, (_T("Failed to get sysinfo error=%d."), errno));
+                    return;
+                }
+                static const long NPROC_ONLN = sysconf(_SC_NPROCESSORS_ONLN);
+                static const float LA_SCALE = static_cast<float>(1 << SI_LOAD_SHIFT);
+                metrics.Set("MemTotal", std::to_string(info.totalram * info.mem_unit));
+                metrics.Set("MemFree", std::to_string(info.freeram * info.mem_unit));
+                metrics.Set("MemSwapped", std::to_string((info.totalswap - info.freeswap) * info.mem_unit));
+                metrics.Set("Uptime", std::to_string(info.uptime));
+                metrics.Set("LoadAvg", std::to_string(info.loads[0] / LA_SCALE) + " " +
+                                 std::to_string(info.loads[1] / LA_SCALE) + " " +
+                                 std::to_string(info.loads[2] / LA_SCALE));
+                metrics.Set("NProc", std::to_string(NPROC_ONLN));
+            };
+            auto parseRssFromStatmLine = [&](const std::string &statmLine, uint32_t &inBytes)
+            {
+                std::vector<std::string> items = splitString(statmLine, ' ');
+                if (items.size() < 7)
+                {
+                    printf("Unexpected size(%u) of 'statm' line.\n", items.size());
```

Comment:
it'd be better to use 'TRACE_GLOBAL(Trace::Error, ...' or similar instead of printf here and other places

Reply:
Yes. Modified the printf logs.